compiler/lexpr.ml

```diff
@@ -265,7 +265,7 @@ let rec tstype trail = function
   | Type_pointer(attr, ty) ->
       tstype (sprintf "*%s" trail) ty
   | Type_array(attr, ty) ->
-      let trail' =
+      let _trail' =
         match attr.bound with
```

Comment:
The warning found a real bug but this is the wrong fix.  `trail'` should be used instead of `trail` in the line following the `None` case.